src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java

```diff
@@ -39,17 +40,14 @@ public void can_invoke_late_assertion_on_assertThatCode() {
   @Test
   public void should_fail_when_asserting_no_exception_raised_but_exception_occurs() {
     // Given
-    ThrowingCallable boom = raisingException("boom");
+    Exception exception = new Exception("boom");
+    ThrowingCallable boom = raisingException(exception);
 
     // Expect
-    thrown.expectAssertionError(format("[Test] %n" +
-                                       "Expecting code not to raise a throwable but caught a%n" +
-                                       "  <java.lang.Exception>%n" +
-                                       "with message :%n" +
-                                       "  \"boom\""));
+    thrown.expectAssertionError(ShouldNotHaveThrown.shouldNotHaveThrown(exception));
 
     // When
-    assertThatCode(boom).as("Test").doesNotThrowAnyException();
+    assertThatCode(boom).doesNotThrowAnyException();
```

Comment:
why removing `as("Test")` ? 
we want to make sure that the error allows a description set with `as`